app/src/main/java/com/android/sample/ui/profile/Profile.kt

```diff
@@ -474,3 +345,333 @@ fun InterestBox(interest: String) {
+                                    .testTag("enrolledActivities")
+                            else Modifier.testTag("enrolledActivities")) {
+                          Icon(Icons.Outlined.Groups, contentDescription = "Enrolled")
+                        }
+                    IconButton(
+                        onClick = { activityType = 2 },
+                        modifier =
+                            if (activityType == 2)
+                                Modifier.background(Color.LightGray, shape = CircleShape)
+                                    .testTag("passedActivities")
+                            else Modifier.testTag("passedActivities")) {
+                          Icon(Icons.Outlined.HourglassFull, contentDescription = "Passed")
+                        }
+                  }
+              Column(
+                  verticalArrangement = Arrangement.spacedBy(STANDARD_PADDING.dp, Alignment.Top),
+                  horizontalAlignment = Alignment.Start,
+                  modifier = Modifier.testTag("activitiesColumn")) {
+                    DisplayActivitiesList(
+                        userActivities,
+                        activityType,
+                        user,
+                        hourDateViewModel,
+                        navigationActions,
+                        profileViewModel,
+                        listActivitiesViewModel,
+                        imageViewModel,
+                        uid)
+                  }
+            }
+      }
+}
+
+@Composable
+fun DisplayActivitiesList(
+    userActivities: List<Activity>,
+    activityType: Int,
+    user: User,
+    hourDateViewModel: HourDateViewModel,
+    navigationActions: NavigationActions,
+    userProfileViewModel: ProfileViewModel,
+    listActivitiesViewModel: ListActivitiesViewModel,
+    imageViewModel: ImageViewModel,
+    uid: String
+) {
+  var listToShow = emptyList<Activity>()
+  when (activityType) {
+    2 -> {
+      listToShow =
+          userActivities.filter {
+            hourDateViewModel.combineDateAndTime(
+                it.date, hourDateViewModel.addDurationToTime(it.startTime, it.duration)) <=
+                Timestamp.now()
+          }
+    }
+    0 -> {
+      listToShow =
+          userActivities.filter {
+            it.creator == user.id &&
+                hourDateViewModel.combineDateAndTime(
+                    it.date, hourDateViewModel.addDurationToTime(it.startTime, it.duration)) >
+                    Timestamp.now()
+          }
+    }
+    1 -> {
+      listToShow =
+          userActivities.filter {
+            (it.creator != user.id || it.participants.map { it.id }.contains(user.id)) &&
+                hourDateViewModel.combineDateAndTime(
+                    it.date, hourDateViewModel.addDurationToTime(it.startTime, it.duration)) >
+                    Timestamp.now()
+          }
+    }
+  }
+  val remainingTime = activityType != 2
```

Comment:
Could you use a constant instead ? It's not very clear to me what activity type 1 or 2 is. A descriptive constant name would definitely help :)

Reply:
Done!